x-pack/plugins/synthetics/public/apps/synthetics/state/monitor_list/api.ts

```diff
@@ -50,13 +51,17 @@ export const fetchDeleteMonitor = async ({ id }: { id: string }): Promise<void>
   return await apiService.delete(`${API_URLS.SYNTHETICS_MONITORS}/${id}`);
 };
 
+export type UpsertMonitorResponse =
+  | { attributes: { errors: ServiceLocationErrors }; id: string }
+  | SavedObject<SyntheticsMonitor>;
+
 export const fetchUpsertMonitor = async ({
   monitor,
   id,
 }: {
   monitor: Partial<SyntheticsMonitor> | Partial<EncryptedSyntheticsMonitor>;
   id?: string;
```

Comment:
```suggestion
  configId?: string;
```
Unrelated to your PR but just for extra clarity.